cybershell/rate_limiter.py

```
@@ -0,0 +1,407 @@
+            
+            # Execute function
+            start_time = time.time()
+            try:
+                result = func(*args, **kwargs)
+                response_time = time.time() - start_time
+                
+                # Adaptive adjustment
+                rate_limiter.adjust_adaptive(response_time, error_occurred=False)
+                
+                return result
+            except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)